programs/clearing_house/src/controller/position.rs

```diff
@@ -184,121 +207,179 @@ pub fn update_position_and_market(
         .ok_or_else(math_error!())?;
 
     // Update Market open interest
-    if new_position {
+    if let PositionUpdateType::Open = update_type {
         market.open_interest = market
             .open_interest
             .checked_add(1)
             .ok_or_else(math_error!())?;
-    } else if closed_position {
+    } else if let PositionUpdateType::Close = update_type {
         market.open_interest = market
             .open_interest
             .checked_sub(1)
             .ok_or_else(math_error!())?;
     }
 
-    if increasing_position {
-        if new_base_asset_amount > 0 {
-            market.base_asset_amount_long = market
-                .base_asset_amount_long
-                .checked_add(delta.base_asset_amount)
-                .ok_or_else(math_error!())?;
-            market.amm.quote_asset_amount_long = market
-                .amm
-                .quote_asset_amount_long
-                .checked_add(delta.quote_asset_amount)
-                .ok_or_else(math_error!())?;
-        } else {
-            market.base_asset_amount_short = market
-                .base_asset_amount_short
-                .checked_add(delta.base_asset_amount)
-                .ok_or_else(math_error!())?;
-            market.amm.quote_asset_amount_short = market
-                .amm
-                .quote_asset_amount_short
-                .checked_add(delta.quote_asset_amount)
-                .ok_or_else(math_error!())?;
+    match update_type {
+        PositionUpdateType::Open | PositionUpdateType::Increase => {
+            if new_base_asset_amount > 0 {
+                market.base_asset_amount_long = market
```

Comment:
since the only difference between the if/else is the _long/short - i think we could make a macro for it and then use '_long/short' suffix as a parameter to the macro - might be overoptim tho

Reply:
i think its a good idea, but probably more legible if we dont add a marco here